src/backend/columnar/columnar_tableam.c

```diff
@@ -1024,13 +1051,16 @@ static void
 columnar_vacuum_rel(Relation rel, VacuumParams *params,
 					BufferAccessStrategy bstrategy)
 {
-	if (!CheckCitusVersion(WARNING))
+	if (ColumnarCheckCitusVersion_hook != NULL)
 	{
-		/*
-		 * Skip if the extension catalogs are not up-to-date, but avoid
-		 * erroring during auto-vacuum.
-		 */
-		return;
+		if (!ColumnarCheckCitusVersion_hook(WARNING))
+		{
+			/*
+			 * Skip if the extension catalogs are not up-to-date, but avoid
+			 * erroring during auto-vacuum.
+			 */
+			return;
+		}
```

Comment:
This comment was quite enlightening.

I guess a lot of the `CheckCitusVersion` invocations were to guard against citus specific functions being called without insurance that the extension had been created and/or upgraded correctly.

How will columnar be handling these cases?

I am asking this to understand if we actually need this hook here, or that columnar will have other guards to make sure the catalogs exist in the correct state.

Given the catalog tables for columnar and citus are sufficiently separated I wonder if we can simple remove the invocations here, or that we need comparable functionality in columnar which will superseed the citus version checks from these call sites.